ui/pages/confirm-add-suggested-nft/confirm-add-suggested-nft.js

```diff
@@ -0,0 +1,296 @@
+import {
+  resolvePendingApproval,
+  rejectPendingApproval,
+} from '../../store/actions';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  MetaMetricsTokenEventSource,
+} from '../../../shared/constants/metametrics';
+import { AssetType } from '../../../shared/constants/transaction';
+import {
+  BUTTON_SIZES,
+  ButtonIcon,
+  ButtonIconSize,
+  ButtonLink,
+  IconName,
+  Text,
+} from '../../components/component-library';
+import {
+  getCurrentChainId,
+  getRpcPrefsForCurrentProvider,
+  getSuggestedNfts,
+  getIpfsGateway,
+} from '../../selectors';
+import NftDefaultImage from '../../components/app/nft-default-image/nft-default-image';
+import { getAssetImageURL, shortenAddress } from '../../helpers/utils/util';
+import {
+  BorderRadius,
+  IconColor,
+  TextAlign,
+  TextVariant,
+} from '../../helpers/constants/design-system';
+import Box from '../../components/ui/box/box';
+
+const ConfirmAddSuggestedNFT = () => {
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+  const suggestedNfts = useSelector(getSuggestedNfts);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const chainId = useSelector(getCurrentChainId);
+  const ipfsGateway = useSelector(getIpfsGateway);
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const handleAddNftsClick = useCallback(async () => {
+    await Promise.all(
+      suggestedNfts.map(async ({ requestData: { asset }, id }) => {
+        await dispatch(resolvePendingApproval(id, null));
+
+        trackEvent({
+          event: MetaMetricsEventName.NftAdded,
+          category: MetaMetricsEventCategory.Wallet,
+          sensitiveProperties: {
+            token_symbol: asset.symbol,
+            token_id: asset.tokenId,
+            token_contract_address: asset.address,
+            source_connection_method: MetaMetricsTokenEventSource.Dapp,
+            token_standard: asset.standard,
+            asset_type: AssetType.NFT,
+          },
+        });
+      }),
+    );
+    history.push(mostRecentOverviewPage);
+  }, [dispatch, history, trackEvent, mostRecentOverviewPage, suggestedNfts]);
+
+  const handleCancelNftClick = useCallback(async () => {
+    await Promise.all(
+      suggestedNfts.map(async ({ id }) => {
+        return dispatch(
+          rejectPendingApproval(
+            id,
+            serializeError(ethErrors.provider.userRejectedRequest()),
+          ),
+        );
+      }),
+    );
+    history.push(mostRecentOverviewPage);
+  }, [dispatch, history, mostRecentOverviewPage, suggestedNfts]);
+
+  const goBackIfNoSuggestedNftsOnFirstRender = () => {
+    if (!suggestedNfts.length) {
+      history.push(mostRecentOverviewPage);
+    }
+  };
+
+  useEffect(() => {
+    goBackIfNoSuggestedNftsOnFirstRender();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should not get in the habit of disabling this check. Either use `useCallback` to prevent the function from being new each render, or move the body of the function into the effect. I think you're trying to ensure that this is only done on the *first render/first call* of the effect. It is safer to use a ref and set the current value of the ref after the effect is ran and then use its value to determine if routing should occur on future runs of the effect.

Reply:
fixed [here](https://github.com/MetaMask/metamask-extension/pull/19454/commits/19234ca62e07ac3b5a83f7181f4f946e68100b43)